kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/LikeExpressionEvaluator.java

```diff
@@ -135,11 +139,21 @@ public boolean isNullAt(int rowId) {
 
             public boolean isLike(String input, String pattern, char escape) {
                 if (!Objects.isNull(input) && !Objects.isNull(pattern)) {
-                    String regex = escapeLikeRegex(pattern, escape);
+                    String regex = getRegexFromCacheOrEval(pattern, escape);
                     return input.matches(regex);
                 }
                 return false;
             }
+
+            public String getRegexFromCacheOrEval(String pattern, char escape) {
+                String regex = (regexCache != null) ?
+                        regexCache :
+                        escapeLikeRegex(pattern, escape);
+                if(isPatternLiteralType) { // set cache only for literals to avoid re-computation
+                    regexCache = regex;
+                }
```

Comment:
```suggestion
                if (regexCache != null) {
                    return regexCache;
                }
                String regex = escapeLikeRegex(pattern, escape);
                if(isPatternLiteralType) { // set cache only for literals to avoid re-computation
                    regexCache = regex;
                }
```